analyzers/tests/SonarAnalyzer.UnitTest/TestCases/LockedFieldShouldBeReadonly.cs

```diff
@@ -214,3 +214,16 @@ namespace ANamespace
 {
     class AClass { }
 }
+
+// https://github.com/SonarSource/sonar-dotnet/issues/7058
+class Repro_7058
+{
+    void Test(int[] a)
+    {
+        var local = new object();
+        Action action = () =>
+        {
+            lock (local) { } // Noncompliant, FP: local is in the captured scope
+        };
```

Comment:
Add a counter example where there is an action, local is captured, but the lock is outside the delegate:
```cs
        var local = new object();
        Action action = () =>
        {
            local.ToString();
        };
        lock (local) { } // Noncompliant
```